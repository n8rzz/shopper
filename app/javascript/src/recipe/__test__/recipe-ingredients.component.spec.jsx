import React from 'react';
import { mount, shallow } from 'enzyme';
import {
    itemMapMock,
    recipeIngredientMock,
} from '../../__mocks__/mocks';
import { RecipeIngredientsComponent } from '../recipe-ingredients.component';

describe('RecipeIngredientsComponent', () => {
    describe('when called without `#ingredients`', () => {
        test('renders correctly with valid props', () => {
            const component = mount(<RecipeIngredientsComponent
                ingredients={[]}
                itemsById={itemMapMock}
            />);

            expect(component.exists('.island')).toEqual(false);
        });
    });

    describe('when called with `#ingredients`', () => {
        test('renders correctly with valid props', () => {
            const component = shallow(<RecipeIngredientsComponent
                ingredients={[recipeIngredientMock]}
                itemsById={itemMapMock}
            />);

            expect(component).toMatchSnapshot();
        });
    });
});
