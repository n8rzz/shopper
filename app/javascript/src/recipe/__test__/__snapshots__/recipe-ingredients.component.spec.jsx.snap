// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeIngredientsComponent when called with \`#ingredients\` renders correctly with valid props 1`] = `
<div
  className="recipeIngredients"
>
  <ul
    className="recipeIngredients-list"
  >
    <li
      className="recipeIngredients-list-item"
      key="ingredient-0"
    >
      <ul
        className="hlist"
      >
        <li>
          10
           
          <span>
            Monkeys
          </span>
        </li>
        <li>
          Basil
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
